cpp/src/arrow/compute/exec/plan_test.cc

```diff
@@ -144,7 +144,14 @@ TEST(ExecPlan, DummyStartProducing) {
   // Note that any correct reverse topological order may do
   ASSERT_THAT(t.started, ElementsAre("sink", "process3", "process2", "process1",
                                      "source2", "source1"));
-  ASSERT_EQ(t.stopped.size(), 0);
+
+  plan->StopProducing();
+  plan->finished().Wait();
```

Comment:
There are some helpful macros for working with futures in `future_util.h`.  You might prefer `ASSERT_FINISHES_OK` so a bug doesn't hang indefinitely

Reply:
I've updated the tests to reuse the default wait